momepy/functional/_diversity.py

```diff
@@ -1,11 +1,97 @@
 import warnings
 
 import numpy as np
+import pandas as pd
 from libpysal.graph import Graph
 from numpy.typing import NDArray
+from packaging.version import Version
 from pandas import DataFrame, Series
 
-__all__ = ["describe"]
+throw_numba_warning = False
+try:
+    from numba import njit
+except (ModuleNotFoundError, ImportError):
+    throw_numba_warning = True
+    from libpysal.common import jit as njit
+
+__all__ = ["describe", "describe_reached"]
+
+
+@njit
+def _mode(array):
+    """Custom mode function for numba."""
+    array = np.sort(array.ravel())
+    mask = np.empty(array.shape, dtype=np.bool_)
+    mask[:1] = True
+    mask[1:] = array[1:] != array[:-1]
+    unique = array[mask]
+    idx = np.nonzero(mask)[0]
+    idx = np.append(idx, mask.size)
+    counts = np.diff(idx)
+    return unique[np.argmax(counts)]
+
+
+@njit
+def _describe(values, q, include_mode=False):
+    """Helper function to calculate average."""
+    nan_tracker = np.isnan(values)
+
+    if (len(values) > 2) and (not nan_tracker.all()):
+        if nan_tracker.any():
+            lower, higher = np.nanpercentile(values, q)
+        else:
+            lower, higher = np.percentile(values, q)
+        values = values[(lower <= values) & (values <= higher)]
+
+    results = [
+        values.shape[0],
+        np.mean(values),
+        np.median(values),
+        np.std(values),
+        np.min(values),
+        np.max(values),
+        np.sum(values),
```

Comment:
Needs to work properly with missing values.

Reply:
I updated the code, I think we can just get rid of the nans beforehand and compute everything on the non_na values, what do you think?